### PR TITLE
Improve test coverage for graph class

### LINKED ISSUE #6102: Improve test coverage for graph class
I discovered there are still some red spots in this link https://app.codecov.io/gh/networkx/networkx/blob/main/networkx/classes/graph.py and will like to work on it.

The tests will be added in https://github.com/networkx/networkx/blob/main/networkx/classes/tests/test_graph.py

Current Behavior
We don't test all the paths the code can take us.

Expected Behavior
We should be testing everything so there aren't any surprises.

Steps to Reproduce
Visit https://app.codecov.io/gh/networkx/networkx/blob/main//networkx/classes/graph.py

### PR DESCRIPTION
**Pull request by Qudirah**
fixes #6102

### CODE REVIEW

**Comment by rossbar** [comment]
Thanks @Qudirah , it looks like a modification to the dispersion tests also snuck into this PR - based on the related issue I assume that was by accident. It'd probably be best to split out the dispersion test updates into a separate branch so that it's clear what this PR is intended to fix.

**Comment by Qudirah** [comment]
@rossbar I already removed the modified files

File: networkx/classes/tests/test_graph.py [hunk 1]
```diff
@@ -683,6 +683,9 @@ def test_add_edge(self):
         G = self.Graph()
         G.add_edge(*(0, 1))
         assert G.adj == {0: {1: {}}, 1: {0: {}}}
+        G = self.Graph()
+        with pytest.raises(ValueError):
+            G.add_edge(None, "anything")
```
Comment (MridulS) [inline networkx/classes/tests/test_graph.py:688]: thanks @Qudirah!

Could you also add this test to the `add_edges_from` function too? That will cover all the other untested code in graph.py.

Comment (Qudirah) [reply networkx/classes/tests/test_graph.py:688]: done @MridulS

**Review by MridulS — APPROVED** [review]
Thanks @Qudirah!

**Pull request merged**